src/transformers/models/gemma2/modeling_gemma2.py

```diff
@@ -807,7 +807,19 @@ def forward(
             inputs_embeds = self.embed_tokens(input_ids)
 
         if cache_position is None:
-            cache_position = torch.arange(0, inputs_embeds.shape[1], device=inputs_embeds.device)
+            if past_key_values is None:
+                cache_position = torch.arange(0, inputs_embeds.shape[1], device=inputs_embeds.device)
+            else:
+                raise ValueError("When `past_key_values` is passed, `cache_position` must be too")
+
+        # Probably a forward call with caching, so we set up cache for one call only
+        if use_cache and past_key_values is None and not self.training:
+            logger.warning_once(
+                "You are calling the model with `use_cache=True` but didn't pass `past_key_values`. ",
+                "Make sure to pass an instance of `HybridCache`. Caching will be disabled. See for more: "
+                "(https://huggingface.co/docs/transformers/main/en/internal/generation_utils#transformers.HybridCache)",
+            )
```

Comment:
```suggestion
                "You are calling the model with `use_cache=True` but didn't pass `past_key_values` while not training. ",
                "If you want to compute with cache, make sure to pass an instance of `HybridCache`. Caching will be disabled otherwise. See for more: "
                "(https://huggingface.co/docs/transformers/main/en/internal/generation_utils#transformers.HybridCache)",
            )
```